### PR TITLE
Feature/localstorage

### PR DESCRIPTION
**Pull request by CorneliusMoe**
added bug fixes and made the input fields required. also added an option to choose between text and date for the last input field and formatted the date to ddmmyyyy

Thanks for reviewing:)

Edit: Fixed the bugs, so the workaround is changed back to initial state and I am a very happy coder.

### CODE REVIEW

File: components/GoalDetailCard/index.js [hunk 1]
```diff
@@ -53,6 +53,20 @@ export default function GoalDetailCard({ goal, deleteGoal }) {
...
 
+  function formatTimelyDate(timely) {
+    const isDate = !isNaN(Date.parse(timely));
+
+    if (isDate) {
+      const date = new Date(timely);
+      const day = date.getDate().toString().padStart(2, "0");
+      const month = (date.getMonth() + 1).toString().padStart(2, "0");
+      const year = date.getFullYear();
+      return `${day}.${month}.${year}`;
```
Comment (marcbendixen) [inline components/GoalDetailCard/index.js:64]: If you want to get the current date in that format you could also try the integrated `toLocaleDateString`like this ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString?retiredLocale=de)):

```suggestion
      const date = new Date().toLocaleDateString('de-DE', { dateStyle: "medium" })
      return date;
```

File: pages/_app.js [hunk 2]
```diff
@@ -1,21 +1,39 @@
...
+  const [allGoals, setGoals] = useState([]);
+
+  useEffect(() => {
+    const storedGoals = localStorage.getItem("goals");
+    if (storedGoals) {
+      setGoals(JSON.parse(storedGoals));
+    } else {
+      setGoals(goals);
+    }
+  }, []);
```
Comment (marcbendixen) [inline pages/_app.js:16]: Just a reminder if you are getting any problems with localStorage in the future: you could use the package `use-local-storage-state` to make it more stable. [See this handout](https://github.com/neuefische/bo-web-23-3/blob/main/sessions/react-with-local-storage/react-with-local-storage.md#react-with-local-storage-1)

**Review by marcbendixen — APPROVED** [review]
Good job 🚀 
I made just a few comments you could keep in mind.
Otherwise LGTM

File: pages/_app.js (hunk 2 above)
Comment (CorneliusMoe) [reply pages/_app.js:16]: Thanks, I was looking through handouts yesterday and somehow managed to completely miss that. I will keep it in mind!

File: components/GoalDetailCard/index.js (hunk 1 above)
Comment (CorneliusMoe) [reply components/GoalDetailCard/index.js]: I decided to do this instead `import { format } from "date-fns";`

**Pull request merged**